google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -300,6 +301,51 @@ public SeekableByteChannel newByteChannel(
     }
   }
 
+  /**
+   * Open a file for reading OR writing. The {@link FileChannel} that is returned will only allow
+   * reads or writes depending on the {@link OpenOption}s that are specified. If any of the
+   * following have been specified, the {@link FileChannel} will be write-only: {@link
+   * StandardOpenOption#CREATE}
+   *
+   * <ul>
+   *   <li>{@link StandardOpenOption#CREATE}
+   *   <li>{@link StandardOpenOption#CREATE_NEW}
+   *   <li>{@link StandardOpenOption#WRITE}
+   *   <li>{@link StandardOpenOption#TRUNCATE_EXISTING}
+   * </ul>
+   *
+   * In all other cases the {@link FileChannel} will be read-only.
+   *
+   * @param path The path to the file to open or create
+   * @param options The options specifying how the file should be opened, and whether the {@link
+   *     FileChannel} should be read-only or write-only.
+   * @param attrs (not supported, the values will be ignored)
```

Comment:
Why include attrs here at all if this is the case? attrs is optional in File.createFile

Reply:
I chose to do so to keep it consistent with the other methods in `CloudStorageFileSystemProvider` that accept an array of `FileAtribute`s as an input parameter. These do the exact same check, even though the input value is ignored.

(See `newByteChannel(...)``createDirectory(...)`)

Furthermore, `newFileChannel` method can also be used to read an existing file, so it does not always call File.createFile.